one/src/main.rs

```diff
@@ -226,25 +233,40 @@ impl Daemon {
         debug!(?p2p_config, "using p2p config");
 
         // Construct a recon implementation.
-        let recon = Arc::new(Mutex::new(ReconEvents::new(BTreeStore::default())));
+        let recon_interest = Arc::new(Mutex::new(Recon::new(
+            BTreeStore::default(),
+            FullInterests::default(),
+        )));
+        let recon_model = Arc::new(Mutex::new(Recon::new(
+            BTreeStore::default(),
+            // Use recon_interest as the InterestProvider for recon_model
+            recon_interest.clone(),
```

Comment:
Given the multiple locks involved we should run this through https://lib.rs/crates/cargo-check-deadlock or the tool tokio uses (can't remember the name of it).

Reply:
I tried using `cargo-check-deadlock` however it fails to compile in its latest release. While the idea is good, I don't think we need CI tooling for deadlocks yet. We only have two locks and they follow basic usage patterns. Its not possible for  deadlock currently as there is no cycle in the code between the locks. 

Additionally I expect we will be moving away from the locks in the medium terms as we look to make Recon more async.